pvanalytics/quality/outliers.py

```diff
@@ -124,3 +126,233 @@ def hampel(data, window=5, max_deviation=3.0, scale=None):
+                                                   gcr=0.4, max_angle=60)
+        surface_tilt = tracker_angles['surface_tilt']
+        surface_azimuth = tracker_angles['surface_azimuth']
+    else:
+        surface_tilt = tilt
+        surface_azimuth = azimuth
+
+    poa = pvlib.irradiance.get_total_irradiance(
+        surface_tilt, surface_azimuth,
+        solar_zenith,
+        solar_azimuth,
+        dni, ghi, dhi,
+        dni_extra=dni_extra,
+        airmass=relative_airmass,
+        albedo=0.2,
+        model='perez'
+    )
+
+    aoi = pvlib.irradiance.aoi(surface_tilt, surface_azimuth,
+                               solar_zenith, solar_azimuth)
+    # Run IAM model
+    iam = pvlib.iam.physical(aoi, n=1.5)
+    # Apply IAM to direct POA component only
+    poa_transmitted = poa['poa_direct'] * iam + poa['poa_diffuse']
+    temp_cell = pvlib.temperature.sapm_cell(
+        poa['poa_global'],
+        temperature,
+        wind_speed,
+        **temperature_model_parameters
+    )
+    pdc = pvlib.pvsystem.pvwatts_dc(
+        poa_transmitted,
+        temp_cell,
+        dc_capacity,
+        temperature_coefficient
+    )
+    return pdc
+
+
+def get_anomalous_days(power_time_series, lat, long, tilt, azimuth,
+                       tracking, nsrdb_weather_df, pct_threshold=50,
+                       dc_capacity=None):
+    """Identifies anomalous days for inverter power time series.
+
+    Compares the actual daily inverter power time series data with PVWatts
+    model power time series output by calculating their percent deference.
+    Days where the difference over the specified percent difference threshold
+    are flagged as anomalous.
+
+    Parameters
+    ----------
+    power_time_series : Pandas series with datetime index.
+        Time series of a PV power data stream with UTC tz-datetime aware index.
+    lat : Float
+        Latitude value of site.
+    long : FLoat
+        Longitude value of site.
+    tilt : Float
+        Tilt angle of site in degrees.
+    azimuth : Float
+        Azimuth angle of site in degrees.
+    tracking : Boolean
+        True if tracking. Otherwise, False if not tracking.
+    nsrdb_weather_df : Pandas dataframe with datetime index
+        A pandas dataframe containing NSRDB PSM3 weather 'Temperature', 'DHI',
+        'DNI', and 'Wind Speed' columns with datetime index. The minimum
+        and maximum datetime matches the minimum and maximum datetime of the
+        actual data.
+    pct_threshold : Float
+        Percent difference threshold for flagging data as anomalies.
+        Defaulted to 50.
+    dc_capacity : None or Float
+        DC capacity of the site. If the inverter dc capacity is not
+        known, set as None so that it can be calculated by finding the 95%
+        percentile of the passed time series.
+        Defaulted to None.
+
+    Returns
+    -------
+    master_df : Pandas dataframe with datetime index
```

Comment:
rename master_df as it's generic

Reply:
Return pandas series of percent difference, add new function to determine if anomalous where output is boolean with datetime index